Makefile

```diff
@@ -10,7 +10,7 @@ clean_pyc:
 	-find . -name '*.py[co]' -exec rm {} \;
 
 build: clean_pyc
-	python setup.py build_ext --inplace
+	python setup.py build_ext --inplace --compiler=$CC
```

Comment:
this is not used in our CI at all, though I do use it locally, so nice to improve this generally.

We should add some documentation (in contributing.rst) that things exist & work (not 100% sure this actually works on windows; the Makefile)

Reply:
That is true, unless ``pandas`` is using ``mingw`` it will not going to use it, indeed
but this part I am just enforcing it, because ``cython`` usually already uses the ``$CC``